Mathlib/Tactic/NthRewrite.lean

```diff
@@ -17,11 +17,48 @@ namespace Mathlib.Tactic
 
 open Lean Elab Tactic Meta Parser.Tactic
 
-/-- `nth_rewrite` is a variant of `rewrite` that only changes the nth occurrence of the expression
-to be rewritten.
+/-- `nth_rewrite` is a variant of `rewrite` that only changes the `n`ᵗʰ _occurrence_ of the
+expression to be rewritten. `nth_rewrite n [eq₁, eq₂, ..., eqₘ]` will rewrite the `n`ᵗʰ _occurrence_
+of each of the `m` equalities `eqᵢ`in that order. Occurrences are counted beginning with `1`.
+Occurrences are counted beginning with `1`. For example,
+```lean
+example (h : a = 1) : a + a + a + a + a = 5 := by
+  nth_rewrite 2 [h]
+/-
+a: ℕ
+h: a = 1
+⊢ a + 1 + a + a + a = 5
+-/
+```
+Notice that the second occurrence of `a` from the left has been rewritten by `nth_rewrite`.
```

Comment:
"from the left" is not 100% correct in my opinion, maybe you could also add the following example to explain why it is the 2nd occurence and not the 1st:

```
example (a b c : Nat) : (a + b) + c = (b + a) + c := by
  nth_rewrite 2 [Nat.add_comm]
  rfl
```

Reply:
Ah yes, this is true. Technically `nth_rewrite` looks at expression trees. I am not sure how convey this intuitively without referring to the expression tree idea.

Reply:
Fixed it with `order of precedence` and added an example to explain the importance of order of precedence.